core/src/main/java/com/google/zxing/datamatrix/encoder/HighLevelEncoder.java

```diff
@@ -295,26 +295,40 @@ static int lookAheadTest(CharSequence msg, int startpos, int currentMode) {
         int[] intCharCounts = new int[6];
         byte[] mins = new byte[6];
         findMinimums(charCounts, intCharCounts, Integer.MAX_VALUE, mins);
-        int minCount = getMinimumCount(mins);
 
-        if (intCharCounts[ASCII_ENCODATION] < intCharCounts[BASE256_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[C40_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[TEXT_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[X12_ENCODATION]
-            && intCharCounts[ASCII_ENCODATION] < intCharCounts[EDIFACT_ENCODATION]) {
+        if (intCharCounts[ASCII_ENCODATION] + 1 <= intCharCounts[BASE256_ENCODATION]
```

Comment:
Here, "+ 1 <=" is the same as "<" so we can leave it.
Below, is "+ 1 <" intended over <= ? just checking. It's different in one line.
Also I wonder if we could wrap this up in a utility method that just finds the minimum count, but maybe the point is that the logic is slightly different each time